packages/jest-snapshot/src/State.ts

```diff
@@ -74,6 +76,17 @@ export default class SnapshotState {
     this.unmatched = 0;
     this._updateSnapshot = options.updateSnapshot;
     this.updated = 0;
+
+    this._reinitializeData = () => {
```

Comment:
could we do `this._initialData = data` instead of defining this function at runtime? then `clear()` would be the body of this `_reinitializeData` but with `this._snapshotData = this._initialData`